tests/foreman/cli/test_host.py

```diff
@@ -2544,6 +2544,84 @@ def test_positive_unregister(self):
 
+    @tier3
+    def test_syspurpose_end_to_end(self):
+        """Create a host with system purpose values set by activation key.
+
+        :id: b88e9b6c-2348-49ce-b5e9-a2b9f0abed3f
+
+        :expectedresults: host is registered and system purpose values are correct.
+
+        :CaseLevel: System
+        """
+        # Create an activation key with test values
+        activation_key = make_activation_key({
+            u'purpose-addons': "test-addon1, test-addon2",
+            u'purpose-role': "test-role",
+            u'purpose-usage': "test-usage",
+            u'service-level': "Self-Support",
+            u'lifecycle-environment-id': self.env['id'],
+            u'organization-id': self.org['id'],
+            u'content-view-id': self.content_view['id'],
+        })
+        # Register a host using the activation key
+        self._register_client(
+            activation_key=activation_key, enable_repo=True, auto_attach=True)
+        self.assertTrue(self.client.subscribed)
+        host = Host.info({'name': self.client.hostname})
+        # Assert sytem purpose values are set in the host as expected
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['purpose-addons'],
+            "test-addon1, test-addon2")
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['purpose-role'],
+            "test-role")
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['purpose-usage'],
+            "test-usage")
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['service-level'],
+            "Self-Support")
+        # Change system purpose values in the host
+        Host.update({
+            u'purpose-addons': "test-addon3",
+            u'purpose-role': "test-role2",
+            u'purpose-usage': "test-usage2",
+            u'service-level': "Self-Support2",
+            u'id': host['id'],
+        })
+        host = Host.info({'id': host['id']})
+        # Assert system purpose values have been updated in the host as expected
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['purpose-addons'],
+            "test-addon3")
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['purpose-role'],
+            "test-role2")
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['purpose-usage'],
+            "test-usage2")
+        self.assertEqual(
+            host['subscription-information']['system-purpose']['service-level'],
+            "Self-Support2")
+        # Assert subscriptions present
+        host_subscriptions = ActivationKey.subscriptions({
+            'organization-id': self.org['id'],
+            'id': activation_key['id'],
+            'host-id': host['id'],
+        }, output_format='json')
+        self.assertGreater(len(host_subscriptions), 0)
```

Comment:
Could we make stronger assertion than "subscription returned something"?

Reply:
Hello @mirekdlugosz 

Would you accept an assertion between these two values:
>>> print(self.subscription_name)                                           
Red Hat Satellite Infrastructure Subscription

and this we get from the AK:

    print(host_subscriptions[0]['name'])                                                                     
    Red Hat Satellite Infrastructure Subscription

The self.subscription_name is what the class sets up


I  had a look at other things and nothing seems better:

this is from the AK:

2020-02-18 12:58:20 - robottelo.ssh - INFO - >>> b'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=json activation-key subscriptions --organization-id="82" --id="125" --host-id="66"'
2020-02-18 12:58:23 - robottelo.ssh - INFO - <<< stdout
[
  {
    "ID": 524,
    "Name": "Red Hat Satellite Infrastructure Subscription",
    "Attached": "1 out of 200",
    "Quantity": null,
    "Start Date": "2019-04-03 04:00:00 UTC",
    "End Date": "2020-04-03 03:59:59 UTC",
    "Support": "Premium",
    "Contract": 11899444,
    "Account": 6212943
  }
]

this is from the host:

>>> print(host['subscription-information'])
{'uuid': '5f72edfb-b8b1-4be6-a1cb-395ad8f27e7a', 'last-checkin': '2020-02-18 12:45:01 UTC', 'release-version': '', 'autoheal': 'true', 'registered-to': 'dell-r330-12.gsslab.brq.redhat.com', 'registered-at': '2020-02-18 12:44:28 UTC', 'registered-by-activation-keys': ['l340quw2cC'], 'system-purpose': {'service-level': 'Self-Support2', 'purpose-usage': 'test-usage2', 'purpose-role': 'test-role2', 'purpose-addons': 'test-addon3'}}

Reply:
Yes, "`subscriptions` includes subscription that we specifically know should be there" sounds like good assertion.

"host was registered using activation key we created a while ago" might be a good addition, too. Not sure how this one fits entire workflow here.

Reply:
we can already be sure it was registered using the key we create in this test because only our key has those syspurpose values.